### PR TITLE
Prototypical batching for GA

### PR DESCRIPTION
**Pull request by alfoa**
--------
Pull Request Description
--------
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Related to #1402 (but not fully addressed yet)

##### What are the significant changes in functionality due to this change request?
This PR introduce an initial prototypical method to execute jobs (models' evaluations) in batches. 
This methodology allows Optimizers and Samplers to submit a set of realizations in "batches". The JobHandler
will execute all the jobs and return them (in bulk) once all of them are finished. 

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.  *Tests will be added with later commit.*
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?
- [x] 9. If any test used as a basis for documentation examples (currently found in `raven/tests/framework/user_guide` and `raven/docs/workshop`) have been changed, the associated documentation must be reviewed and assured the text matches the example.

### CODE REVIEW

**Comment by alfoa** [comment]
@joshua-cogliati-inl @mandd @wangcj05

**Comment by alfoa** [comment]
I am still running the tests locally to check if they work but this can be start to be reviewed (from the code prospective)

**Comment by alfoa** [comment]
@joshua-cogliati-inl @mandd this now works.

**Comment by mandd** [comment]
> @joshua-cogliati-inl @mandd this now works.

Awesome, thanks!

**Comment by alfoa** [comment]
@mandd @joshua-cogliati-inl  I think this PR is ready to go (or at least to be reviewed). I think I can modify the "allMathcing" thing directly in the GA branch... At least it is testable in the actualy algorithm

**Review by joshua-cogliati-inl — CHANGES_REQUESTED** [review]
I have some questions and some comments on the pull request.  Thank you for splitting it out, it is more understandable.

File: framework/DataObjects/DataSet.py
Comment (alfoa) [reply framework/DataObjects/DataSet.py:1578]: fixed

File: framework/DataObjects/DataSet.py
Comment (alfoa) [reply framework/DataObjects/DataSet.py:1579]: fixed

File: framework/Steps.py
Comment (alfoa) [reply framework/Steps.py:780]: fixed

File: framework/JobHandler.py
Comment (alfoa) [reply framework/JobHandler.py:559]: commented

File: framework/JobHandler.py
Comment (alfoa) [reply framework/JobHandler.py:609]: I think it was a debug statement.... removed

Comment (alfoa) [reply framework/JobHandler.py:609]: I think it was a debug statement...removed

File: framework/JobHandler.py
Comment (alfoa) [reply framework/JobHandler.py:349]: documented

File: framework/Steps.py
Comment (alfoa) [reply framework/Steps.py:685]: commented

File: framework/Steps.py
Comment (alfoa) [reply framework/Steps.py:730]: commented

**Comment by alfoa** [comment]
@joshua-cogliati-inl  I pushed the modifications

File: framework/Steps.py
Comment (joshua-cogliati-inl) [reply framework/Steps.py:738]: Okay, at least there is a fixme comment.

**Review by joshua-cogliati-inl — COMMENTED** [review]
Changes addressed.

**Review by joshua-cogliati-inl — DISMISSED** [review]

File: framework/JobHandler.py
Comment (alfoa) [reply framework/JobHandler.py]: addressed

File: framework/Models/Model.py
Comment (joshua-cogliati-inl) [inline framework/Models/Model.py:140]: Adding GeneticAlgorithm probably should wait until GA is merged.

**Review by joshua-cogliati-inl — APPROVED** [review]
I have reviewed the changes and I approve.

File: framework/Models/Model.py
Comment (alfoa) [reply framework/Models/Model.py:140]: addressed

**Comment by alfoa** [comment]
@joshua-cogliati-inl  i addressed last comment

**Comment by mandd** [comment]
i had invalidated again , just to make sure it is a coincidence

**Comment by mandd** [comment]
The passed now

**Pull request merged**
